### ISSUE TITLE
oidentd fails to drop privileges if they are already dropped

### ISSUE THREAD

**Issue by Nicd** [janikrabe/oidentd#8]
Sometime recently oidentd became unable to function if it was started with restricted privileges. I used to run xinetd with the config:

```
user = nobody
server = oidentd
server_args = -I
```

As far as I know this would start oidentd as `nobody` and oidentd would then listen to STDIO for the request that comes from xinetd. But now oidentd fails to start, complaining:

```
Fatal: Failed to drop privileges (global)
```

But if you listen from STDIO, you don't need to try to drop privileges if you are already non-root, no? You can easily test this by running oidentd as your own user:

```
➜  ~ oidentd -I
Fatal: Failed to drop privileges (global)
```

I had to change my xinetd config to say `user = root` and `server_args = -I -u nobody`, so now I am starting oidentd as root only for it to drop back to nobody immediately.

Is this an issue? I would have expected oidentd only to drop privileges if it's running as root in the first place.

**Comment by janikrabe** [comment]
Hi, thanks for reporting! You're right, this doesn't look like something that should be happening. I'll take a look later.

Which operating system are you using?

**Comment by Nicd** [comment]
Ran into this on an amd64 Arch Linux system.

**Comment by janikrabe** [comment]
Thanks for your report! This issue has been fixed in 4ca63dd46c937ff4a97986a912e93bcab8cd62bf.
